handler/schedule.go

```diff
@@ -125,3 +129,43 @@ func (s *ScheduleHandler) mergeResourcesWithDefaults(resources corev1.ResourceRe
 	}
 	return resources
 }
+
+func (s *ScheduleHandler) updateSchedule() error {
+	if err := s.Client.Update(s.CTX, s.schedule); err != nil && !errors.IsNotFound(err) {
+		return fmt.Errorf("error updating resource %s/%s: %w", s.schedule.Namespace, s.schedule.Name, err)
+	}
+	return nil
+}
+
+func (s *ScheduleHandler) updateStatus() error {
+	err := s.Client.Status().Update(s.CTX, s.schedule)
+	if err != nil {
+		s.Log.Error(err, "Could not update SyncConfig.", "name", s.schedule)
+		return err
+	}
+	s.Log.Info("Updated SyncConfig status.")
+	return nil
+}
+
+func (s *ScheduleHandler) getEffectiveSchedule(jobType k8upv1alpha1.JobType, originalSchedule string) string {
+	if s.schedule.Status.EffectiveSchedules == nil {
+		s.schedule.Status.EffectiveSchedules = make(map[k8upv1alpha1.JobType]string)
+	}
+	if existingSchedule, found := s.schedule.Status.EffectiveSchedules[jobType]; found {
+		return existingSchedule
+	}
+	if !strings.HasSuffix(originalSchedule, "-random") {
+		return originalSchedule
+	}
+	schedule := originalSchedule
+	seed := s.schedule.Namespace + "/" + s.schedule.Name + "@" + string(jobType)
+	schedule, err := randomizeSchedule(seed, originalSchedule)
+	if err != nil {
+		s.Log.Info("Could not randomize schedule, ignoring and try original schedule", "seed", seed, "error", err.Error())
+	} else {
+		s.Log.V(1).Info("Randomized schedule", "seed", seed, "from_schedule", originalSchedule, "effective_schedule", schedule)
```

Comment:
I'm wondering what value the `seed` information has to an outside observer. I would say it's more confusing that helping when debugging, because it's an internal information, and therefore not add it to the log statements (here and the one above).

Reply:
I would argue that I need internal information when debugging, otherwise debugging would be fruitless, right?
Given that the stable randomization with seed is a requested feature, and the seed is relevant for the randomization, I see no harm when logging this. Maybe someone finds an issue with randomized schedules ("they are all the same" etc), then the seed becomes very much relevant.
For the Info log above, I agree though